### ISSUE TITLE
Missing values in summary files

### ISSUE THREAD

**Issue by AlexKurek** [lofar-astron/prefactor#267]
This is the end of summary file for both calibrator and target:

> Overall amount of flagged data in the final data:
> Station  

I can also see a lot of such warnings in both callibrator and target logs:

> 2020-04-13 11:24:05 WARNING node.mpuls.executable_args.L232873_SB000_uv.dppp.MS: /opt/lofar/bin/NDPPP stderr: 
> *** WARNING: the following parset keywords were not used ***
>              maybe they are misspelled
>     [aoflag.keepstatistics,aoflag.memoryperc,aoflag.type,demix.baseline,demix.demixfreqstep,demix.demixtimestep,demix.freqstep,demix.ignoretarget,demix.instrumentmodel,demix.ntimechunk,demix.skymodel,demix.subtractsources,demix.targetsource,demix.timestep,demix.type,flagedge.chan,flagedge.type]

Is this expected? The image generated by image-parset looks OK though.

Prefactor commit: 085e39875592e6b883327714a56ebd8be5abdf5d

Logs and summary files are attched. I had to stop computation at some point, this is why there are two target log-s.
[log.zip](https://github.com/lofar-astron/prefactor/files/4496340/log.zip)
[summary.zip](https://github.com/lofar-astron/prefactor/files/4496342/summary.zip)

**Comment by adrabent** [comment]
Dear @AlexKurek 
I recently run the most recent version of prefactor and could not reproduce your error.
In your logfile I saw this error message while running the summary-script:

`DPPP: error while loading shared libraries: libarmadillo.so.9: cannot open shared object file: No such file or directory`

Armadillo is one of the LOFAR software requirements and seems to be missing if the script aims for running DPPP. If your DPPP is broken, you should not have been able to run prefactor at all. The summary scripts calls DPPP from the "command line". Please check in your terminal with `which DPPP` whether this matches `/opt/lofar/bin/NDPPP`.

Alex

**Comment by AlexKurek** [comment]
Yes, it matches this path.

I added LD_LIBRARY_PATH for Armadillo and calibrators summary file looks OK now (attached). But not used parsets keywords are still in the logs. I went back to Prefactor 3.0 and it didnt helped. I also tried my much older instalation scripts (they are not checking out any version of Prefactor / DP3 etc.) and not used keywords are still there too. Now I will go back to older DP3 and see if it helps.
[Pre-Facet-Calibrator.zip](https://github.com/lofar-astron/prefactor/files/4508163/Pre-Facet-Calibrator.zip)

**Comment by adrabent** [comment]
> NDP3 version is 3.2.18, DP3 version is 4.1
What is the difference between NDP3 and DP3? Shouldn't it be the same?

**Comment by tikk3r** [comment]
> Yes, it matches this path. NDP3 version is 3.2.18, DP3 version is 4.1



> What is the difference between NDP3 and DP3? Shouldn't it be the same?

NDP3 probably refers tot the one that is installed when you install the LOFAR stack. To avoid this, I typically disable building DP3 in the LOFAR software and then symlink lofar/bin/NDPPP to wherever the now standalone DPPP is installed.

(Unless this has changed over time and the standalone DP3 is not included in the LOFAR stack.)

> But not used parsets keywords are still in the logs.

This is typically fine. That means that you are not using the respective `aoflag`, `demix` and `flagedge` steps. These can be enabled, if you want, by changing the `{{ initial_flagging }}` and `{{ demix }}` entries in the parset.

**Comment by AlexKurek** [comment]
> NDP3 probably refers tot the one that is installed when you install the LOFAR stack.

Yes, it does

> To avoid this, I typically disable building DP3 in the LOFAR software

Do you mean `-DBUILD_Imager=OFF` in LOFARsoft?

**Comment by tikk3r** [comment]
> Do you mean `-DBUILD_Imager=OFF` in LOFARsoft?

Imager is something else I believe, maybe `-DBUILD_DP3=OFF`, but that's a guess on my side (I only know that DP3 can be passed to -DBUILD_PACKAGES, so maybe the opposite works as well). I work the other way around myself, using `-DBUILD_PACKAGES` to only select what is needed for the pipeline.

**Comment by aroffringa** [comment]
Good idea would probably be to remove DP3 from the LOFAR repository. @tammojan any reason for DP3 to be still in LOFAR?

**Comment by tammojan** [comment]
No valid reason, I'll re-raise this with the maintainers of the LOFAR repo.

**Comment by AlexKurek** [comment]
$tikk3r, could you please paste the list of packages needed by the pipeline? I will try to compile this way

**Comment by tikk3r** [comment]
> @tikk3r, could you please paste the list of packages needed by the pipeline? I will try to compile this way

I use `-DBUILD_PACKAGES="StationResponse pystationresponse ParmDB pyparmdb Pipeline MS"`. You can see my full setup here: https://github.com/tikk3r/lofar-grid-hpccloud/blob/master/Singularity.lofar_sksp

**Comment by AlexKurek** [comment]
@tikk3r, thank you, this is very interesting! I have two questions:
- why are you building station response and python station response? As far as I understand this is branched off now and has to be installed from https://github.com/lofar-astron/LOFARBeam anyway
- I can see that you disabled log4cplus. Is this not essential for producing Prefactors logs?

**Comment by tikk3r** [comment]
>     * why are you building station response and python station response?

Probably I just left it in. I do build LOFARBeam later on and add that to all the (build) paths.


>     * I can see that you disabled log4cplus. Is it not essential for producing Prefactors logs?

Logs work just fine without it for me. I used to constantly see warnings in the pipeline about it not being supported or used, so I disabled it.


>     * f I compile only those packages, cmake says that `DCASAREST_ROOT_DIR` was not used. So maybe Prefactor do not need Casarest at all?

I don't build casarest even. As far as I know it gives lwimager, which isn't used anymore.

The symlinking of NDPPP to DPPP I did, because the genericpipeline seems to be hardcoded to call $LOFARROOT/bin/NDPPP (or I'm missing some configuration setting somewhere). That was the easiest way for me to get it to pick the right DP3.

**Comment by AlexKurek** [comment]
Thank you. I compilled everything your way and its computing now. I left only `-DBUILD_PACKAGES="ParmDB pyparmdb Pipeline MS"` and this seems also OK for now.

`ulimit -n 4096` was not working for me in `runscript`, but I moved it to `enviroment `like you did and now its working.

I attach my Singularity recipes. Maybe you will find it useful since I have more recent versions of some components and I tested that its still working.
[SING_pref3_ubuntu18_python2.zip](https://github.com/lofar-astron/prefactor/files/4515555/SING_pref3_ubuntu18_python2.zip)
